tests/process_input_test.py

```diff
@@ -0,0 +1,38 @@
+"""Tests for the process_input Module."""
+from agent import process_input
+
+
+def testProcessLink_Always_returnCorrectType():
+    github_links = [
+        "https://github.com/user/repo.git",
+        "http://github.com/Ostorlab/agent_trufflehog.git",
+    ]
+    gitlab_links = [
+        "https://gitlab.com/kalilinux/packages/trufflehog.git",
+        "http://gitlab.com/open-source-projects-lambda/cpython_mirror.git",
+    ]
+    other_links = [
+        "http://www.example.com/",
+        "http://www.example.edu/ball/box.html",
+        "https://sample.info/?insect=fireman&porter=attraction#cave",
+    ]
```

Comment:
Use parameterized tests to run these.

Reply:
[65ededf](https://github.com/Ostorlab/agent_trufflehog/pull/2/commits/65ededff9deb0aab9ac7133b6049a7a0e3034dff) changed.